ntp.go

```diff
@@ -38,6 +38,15 @@ func (t ntpTime) UTC() time.Time {
 	return time.Date(1900, 1, 1, 0, 0, 0, 0, time.UTC).Add(time.Duration(nsec))
 }
 
+func toNtpTime(t time.Time) ntpTime {
+	seconds := uint32(t.Unix() - time.Date(1900, 1, 1, 0, 0, 0, 0, time.UTC).Unix())
+	fraction := t.Nanosecond() << 32
```

Comment:
I don't think you should be using an int at all here - you need to use a float64.